File: src/sanitizeSpecialCharacters.ts
import { mappingToCharacter } from "./characterCodes";

export const sanitizeSpecialCharacters = (text: string): string => {
  return text
    .split("")
    .map(mappingToCharacter)
    .join("")
    .replaceAll("\u2674\uFE0F", "\u2674");
};

File: python/pyvbml/sanitize_special_characters.py
"""Sanitize special characters.

Port of Vestaboard/vbml/src/sanitizeSpecialCharacters.ts
"""

from __future__ import annotations

from .character_codes import mapping_to_character


def sanitize_special_characters(text: str) -> str:
    """Sanitize special characters."""
    return "".join(mapping_to_character(ch) for ch in text)
